src/main/java/bookstore/mapper/BookMapper.java

```diff
@@ -2,13 +2,39 @@
 
 import bookstore.config.MapperConfig;
 import bookstore.dto.book.BookDto;
+import bookstore.dto.book.BookDtoWithoutCategoryIds;
 import bookstore.dto.book.CreateBookRequestDto;
 import bookstore.model.Book;
+import bookstore.service.impl.CategoryServiceImpl;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.mapstruct.AfterMapping;
 import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.Named;
 
-@Mapper(config = MapperConfig.class)
+@Mapper(config = MapperConfig.class, uses = CategoryServiceImpl.class)
```

Comment:
```suggestion
@Mapper(config = MapperConfig.class)
```

Reply:
changed algorithm of categories initialization while Book create/update requests